kymatio/scattering2d/frontend/base_frontend.py

```diff
@@ -0,0 +1,107 @@
+            The maximum order of scattering coefficients to compute. Must be either
+            `1` or `2`. Defaults to `2`.
+        pre_pad : boolean, optional
+            Controls the padding: if set to False, a symmetric padding is applied
+            on the signal. If set to True, the software will assume the signal was
+            padded externally. Defaults to `False`.
+        backend : object, optional
+            Controls the backend which is combined with the frontend.
+
+        Attributes
+        ----------
+        J : int
+            Log-2 of the scattering scale.
+        shape : tuple of int
+            Spatial support (M, N) of the input.
+        L : int, optional
+            Number of angles used for the wavelet transform.
+        max_order : int, optional
+            The maximum order of scattering coefficients to compute.
+            Must be either equal to `1` or `2`.
+        pre_pad : boolean
+            Controls the padding: if set to False, a symmetric padding is applied
+            on the signal. If set to True, the software will assume the signal was
+            padded externally.
+        Psi : dictionary
+            Contains the wavelets filters at all resolutions. See
+            filter_bank.filter_bank for an exact description.
+        Phi : dictionary
+            Contains the low-pass filters at all resolutions. See
+            filter_bank.filter_bank for an exact description.
+        M_padded, N_padded : int
+             Spatial support of the padded input.
+
+        Notes
+        -----
+        The design of the filters is optimized for the value L = 8.
+
+        pre_pad is particularly useful when cropping bigger images because
+        this does not introduce border effects inherent to padding.
+        """
+        
+    def __init__(self, J, shape, L=8, max_order=2, pre_pad=False,\
+           backend=None):
+        super(Scattering2DBase, self).__init__()
+        self.pre_pad, self.L, self.backend, self.J, self.shape, self.max_order = pre_pad, L, backend, J, shape,\
+                                                                            max_order
+
+    def build(self):
+        self.M, self.N = self.shape
+        # use the default backend if no backend is provided
+
+        if 2 ** self.J > self.shape[0] or 2 ** self.J > self.shape[1]:
+            raise RuntimeError('The smallest dimension should be larger than 2^J.')
+        self.M_padded, self.N_padded = compute_padding(self.M, self.N, self.J)
+        # pads equally on a given side if the amount of padding to add is an even number of pixels, otherwise it adds an extra pixel
+        self.pad = self.backend.Pad([(self.M_padded - self.M) // 2, (self.M_padded - self.M+1) // 2, (self.N_padded - self.N) // 2,
+                                (self.N_padded - self.N + 1) // 2], [self.M, self.N], pre_pad=self.pre_pad)
+        self.unpad = self.backend.unpad
+        
+    def create_filters(self):
```

Comment:
Docstring?

Reply:
isn't it done in the inherited class?

Reply:
I suppose this is true. I guess it seems like we should write down what the errors raised are through, like RuntimeErrors.

Reply:
Go for it! Let's try to avoid redundant documentation

Reply:
I guess this is solved.